### PR TITLE
Button & themeprovider

### PR DESCRIPTION
**Pull request by laurensvc**
new button component 
changes to how theme provider works, this is breaking !
After this pr it is required to import UnbraceThemeProvider in you app ( instead of regular )

### CODE REVIEW

File: src/components/UnbraceThemeProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { ThemeProvider } from 'styled-components';
+import merge = require('lodash.merge');
+import { componentTheme, ComponentTheme } from '../theme';
+
+type Props = {
+  children: React.ReactElement;
+  customTheme?: Partial<ComponentTheme>;
```
Comment (RobbeB) [inline src/components/UnbraceThemeProvider.tsx]: This typing is too restricting for the user. This only allows for their own custom theme to be a partial of our components theme. They should be allowed to have whatever theme they want

File: src/theme/components/globals.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const globals = {
```
Comment (RobbeB) [inline src/theme/components/globals.ts]: Make this the same structure as we discussed

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import styled, { css } from 'styled-components';
+
+export type ButtonProps = {
+  block?: boolean;
+  children?: React.ReactNode;
+  color?: 'primary' | 'secondary' | 'tertiary' | 'danger';
+  disabled?: boolean;
+  isLoading?: boolean;
```
Comment (RobbeB) [inline src/components/Button.tsx]: We should be consistent in naming our boolean props, either prefix all of them with is... or remove it everywhere (isLoading / loading, isDisabled / disabled)

Comment (laurensvc) [reply src/components/Button.tsx]: fixed

File: src/components/UnbraceThemeProvider.tsx (hunk 1 above)
Comment (laurensvc) [reply src/components/UnbraceThemeProvider.tsx]: fixed

File: src/theme/components/globals.ts (hunk 2 above)
Comment (laurensvc) [reply src/theme/components/globals.ts]: fixed

File: src/components/UnbraceThemeProvider.tsx (hunk 1 above)
Comment (RobbeB) [reply src/components/UnbraceThemeProvider.tsx]: This still isn't right

File: src/theme/components/button.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { DefaultTheme } from '..';
+
+const getButtonTheme = (theme: DefaultTheme) => ({
+  background: {
+    danger: theme.palette.danger.shade5,
+    dangerHover: theme.palette.danger.shade6,
```
Comment (THMKAE) [inline src/theme/components/button.ts]: Should hover be a nested property?

`background: {
   primary: ...
   secondary: ...
  ...
   hover: {
      primary: ...
      ...
   }
}`

**Comment by THMKAE** [comment]
Make sure to also update the Readme

**Pull request merged**